packages/tools/examples/labelmapSegmentationDynamicThreshold/index.ts

```diff
@@ -0,0 +1,525 @@
+  await initDemo();
+
+  utilities.imageRetrieveMetadataProvider.add(
+    'volume',
+    ProgressiveRetrieveImages.interleavedRetrieveStages
+  );
+
+  // Add tools to Cornerstone3D
+  cornerstoneTools.addTool(PanTool);
+  cornerstoneTools.addTool(ZoomTool);
+  cornerstoneTools.addTool(StackScrollMouseWheelTool);
+  cornerstoneTools.addTool(StackScrollTool);
+  cornerstoneTools.addTool(SegmentationDisplayTool);
+  cornerstoneTools.addTool(RectangleScissorsTool);
+  cornerstoneTools.addTool(CircleScissorsTool);
+  cornerstoneTools.addTool(SphereScissorsTool);
+  cornerstoneTools.addTool(PaintFillTool);
+  cornerstoneTools.addTool(BrushTool);
+
+  // Define tool groups to add the segmentation display tool to
+  const toolGroup = ToolGroupManager.createToolGroup(toolGroupId);
+
+  // Manipulation Tools
+  toolGroup.addTool(PanTool.toolName);
+  toolGroup.addTool(ZoomTool.toolName);
+  toolGroup.addTool(StackScrollMouseWheelTool.toolName);
+
+  // Segmentation Tools
+  toolGroup.addTool(SegmentationDisplayTool.toolName);
+  toolGroup.addTool(RectangleScissorsTool.toolName);
+  toolGroup.addTool(CircleScissorsTool.toolName);
+  toolGroup.addTool(SphereScissorsTool.toolName);
+  toolGroup.addToolInstance(
+    brushInstanceNames.ScissorsEraser,
+    SphereScissorsTool.toolName,
+    {
+      activeStrategy: brushStrategies.ScissorsEraser,
+    }
+  );
+  toolGroup.addTool(PaintFillTool.toolName);
+  toolGroup.addTool(StackScrollTool.toolName);
+  toolGroup.addToolInstance(
+    brushInstanceNames.CircularBrush,
+    BrushTool.toolName,
+    {
+      activeStrategy: brushStrategies.CircularBrush,
+      previewColors,
+      strategySpecificConfiguration: {
+        useCenterSegmentIndex: true,
+      },
+    }
+  );
+  toolGroup.addToolInstance(
+    brushInstanceNames.CircularEraser,
+    BrushTool.toolName,
+    {
+      activeStrategy: brushStrategies.CircularEraser,
+      previewColors,
+    }
+  );
+  toolGroup.addToolInstance(
+    brushInstanceNames.SphereBrush,
+    BrushTool.toolName,
+    {
+      activeStrategy: brushStrategies.SphereBrush,
+      previewColors,
+    }
+  );
+  toolGroup.addToolInstance(
+    brushInstanceNames.SphereEraser,
+    BrushTool.toolName,
+    {
+      activeStrategy: brushStrategies.SphereEraser,
+      previewColors,
+    }
+  );
+  toolGroup.setToolActive(StackScrollTool.toolName, {
+    bindings: [
+      {
+        mouseButton: MouseBindings.Primary, // Left Click
+        modifierKey: KeyboardBindings.Alt,
+      },
+      {
+        numTouchPoints: 1,
+        modifierKey: KeyboardBindings.Meta,
+      },
+    ],
+  });
+
+  const cancelFn = {
+    fn: (evt) => {
+      const { element, key } = evt.detail;
+      const toolGroup = ToolGroupManager.getToolGroup(toolGroupId);
+      const activeName = toolGroup.getActivePrimaryMouseButtonTool();
+      const brush = toolGroup.getToolInstance(activeName);
+      if (key === 'Escape') {
+        brush.rejectPreview?.(element);
```

Comment:
This is not the cornerstone way of doing things. We should not allow calling methods directly on tool instances. One approach you can take is similar to how the AdvancedMagnify or SplineTool works. They create their own bindings in their constructor for the `action` methods, and then our listeners would invoke the corresponding method. This way, you won't need any of these.

See AdvancedMagnifty

```js
actions: {
  showZoomFactorsList: {
    method: 'showZoomFactorsList',
    bindings: [
      {
        mouseButton: MouseBindings.Secondary,
        modifierKey: KeyboardBindings.Shift,
      },
    ],
  },
},
```

which we handle internally inside `getToolsWithActionsForMouseEvent`

Reply:
That doesn't help - those are mouse bindings you are looking at, with a keyboard modifier, not direct actions which are invokeable externally.  Ideally I'd like the preview accept/reject functions to be external, but I don't know how to set that up, and at this point I'm not willing to create a full framework for keyboard handling to invoke tool functions.
Can we leave this for now, and improve on it during the undo phase?  We will need a way to undo/redo, and also to handle preview capabilities, so if we setup listeners that can be fired externally and the tool responds to when active, then that would be optimal.

Reply:
There is no framework, You need these 60 lines of code for keyboard, right now it is `getToolsWithActionsForMouseEvent` you need an exact copy paste to `getToolsWithActionsForKeyboardEvents` which the `keyDown` should call. 

Calling methods on tool instances is just a no no IMO

Alternative would be a utility that deals with the state